version: "3.9"
   
services:
  # db:
  #   image: postgres
  #   volumes:
  #     - ./data/db:/var/lib/postgresql/data
  #   environment:
  #     - POSTGRES_DB=giakinh0823
  #     - POSTGRES_USER=giakinh0823
  #     - POSTGRES_PASSWORD=giakinh0823
  #   ports:
  #     - "5432:5432"
  #   healthcheck:
  #     test: ["CMD-SHELL", "pg_isready -U postgres"]
  #     interval: 10s
  #     timeout: 5s
  #     retries: 5
  #   restart: always

  redis:
    image: redis:alpine

  celery:
    build: .
    command: bash -c "python manage.py makemigrations && python manage.py migrate && celery -A backend worker -l info"
    volumes:
      - .:/code
    depends_on:
      - web
      - redis

  celery-beat:
    build: .
    command: bash -c "python manage.py makemigrations && python manage.py migrate && celery -A backend beat -l info"
    volumes:
      - .:/code
    depends_on:
      - web
      - redis

  web:
    build: .
    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/code
    ports:
      - "8000:8000"
    environment:
      - POSTGRES_NAME=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - CELERY_BROKER=redis://redis:6379/0
      - CELERY_BACKEND=redis://redis:6379/0
    depends_on:
      - redis

